### PR TITLE
FW-4234 improved admin forms

### PR DESCRIPTION
**Pull request by caforbes**
### Description of Changes

Added improved site-by-site filtering for content on misc admin forms, with particular attention to the Site form and Character forms (addressing some of FW-4234). There is now a site filter in various admin lists. Dropdowns now only contain content from the same site (this fixes WOTD bug FW-4405). The filtered queries improve admin site performance/load times.

### Checklist
- [X] README / documentation has been updated if needed
- [X] PR title / commit messages contain Jira ticket number if applicable
- Tests have been updated / created if applicable
- [X] Admin Panel has been updated if models have been added or modified
- Migrations have been updated and committed if applicable
- Team Postman workspace has been updated if applicable

### Reviewers Should Do The Following:
- [ ] Review the code changes here
- [ ] Pull the branch and test locally

### Additional Notes
N/A

### CODE REVIEW

File: firstvoices/backend/admin/characters_admin.py [hunk 1]
```diff
@@ -74,37 +111,9 @@ class IgnoredCharacterAdmin(BaseSiteContentAdmin):
 
 
 @admin.register(Alphabet)
-class AlphabetAdmin(BaseSiteContentAdmin):
+class AlphabetAdmin(BaseAdmin):
```
Comment (sarahfirstvoices) [inline firstvoices/backend/admin/characters_admin.py:114]: Why the change in base class? It's minor but the BaseSiteContentAdmin does a select_related for the site field that helps when there are a lot of items on the list page

File: firstvoices/backend/admin/sites_admin.py [hunk 2]
```diff
@@ -78,9 +78,9 @@ class SiteFeatureAdmin(BaseSiteContentAdmin):
 
 
 @admin.register(SiteMenu)
-class SiteMenuAdmin(BaseSiteContentAdmin):
+class SiteMenuAdmin(BaseAdmin):
     fields = ("site", "json")
-    list_display = ("json",) + BaseSiteContentAdmin.list_display
+    list_display = ("json", "site") + BaseAdmin.list_display
```
Comment (sarahfirstvoices) [inline firstvoices/backend/admin/sites_admin.py:83]: Same question: any reason for this change?

File: firstvoices/backend/admin/characters_admin.py (hunk 1 above)
Comment (caforbes) [reply firstvoices/backend/admin/characters_admin.py:114]: I changed it (for both) because these models essentially provide 1 object per site, and the only content is a json field. I had updated BaseSiteContentAdmin to let you filter rows per site, and that wasn't necessary when there is only 1 row per site (or less) in the first place. An alternative would be to return this to BaseSiteContentAdmin and remove the filter and duplicate `site` columns, if that would be better.

Comment (sarahfirstvoices) [reply firstvoices/backend/admin/characters_admin.py:114]: Oh, gotcha! I vote for that alternate way, just to make sure we get all the prefetching everywhere

Comment (sarahfirstvoices) [reply firstvoices/backend/admin/characters_admin.py:114]: I'm marking this approved because I think it's fine either way. No need to re-review if you make that change :)

**Review by sarahfirstvoices — APPROVED** [review]
